components/object_store/src/obkv/meta.rs

```diff
@@ -177,8 +177,13 @@ impl ObkvObjectMeta {
         let batch_size = self.part_size;
         let start_index = range.start / batch_size;
         let start_offset = range.start % batch_size;
-        let end_index = range.end / batch_size;
-        let end_offset = range.end % batch_size;
+        let mut end_index = range.end / batch_size;
+        let mut end_offset = range.end % batch_size;
+
+        if end_offset == 0 {
+            end_index -= 1;
```

Comment:
If `range.end == 0`, then this may overflow? And I guess this bug results from the `range.end` is exclusive, so how about putting it as:
```rust
let inclusive_end =. if range.end == 0 {
   return Ok(...)
} else {
   range.end - 1
};

let end_index = inclusive_end / batch_size;
let end_offset = inclusive_end / batch_size;
```

Reply:
Oops, yes, it may overflow here. And I will fix it.